File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int product=1,numOfZeros=0;
        
        for(int i=0;i<nums.Length;i++){
            if(nums[i]!=0)
              product=GetProduct(product,nums[i]);
            else 
            numOfZeros++;
        }
        Console.WriteLine(product+" "+numOfZeros);
        for(int i=0;i<nums.Length;i++){
            if(nums[i]!=0){
                nums[i]=numOfZeros>0?0:product/nums[i];
                continue;
            }
            nums[i]=numOfZeros<=1?product:0;   
        }
        return nums;
    }
    private int GetProduct(int firstNum,int seconNum){
        int product=0;
        bool isPostive=((firstNum+seconNum)/seconNum)>0;
        firstNum=Math.Abs(firstNum);
        seconNum=Math.Abs(seconNum);
        while(firstNum>0){
            int lastNumBitNumber=(firstNum&1);
            if(lastNumBitNumber==1)
                product+=seconNum;
            firstNum>>=1;
            seconNum<<=1;
        }
        return product/(isPostive?1:-1);
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int sz=(int)nums.size();
        vector<int>v(sz);
        int res=1;
        v[0]=1;
        for(int i=0;i<sz;i++)
            v[i]=res,res*=nums[i];
        
        res=1;
        
        for(int i=sz-1;i>=0;i--){
            v[i]=v[i]*res;
            res*=nums[i];
        }
        return v;
    }
};